clients/include/testing_preconditioner.hpp

```diff
@@ -325,7 +325,6 @@ void testing_preconditioner_all(Arguments argus)
     e.Allocate("e", matrix.GetM());
 
     ls.MoveToAccelerator();
-    p->MoveToAccelerator();
```

Comment:
Why is this required? From a users perspective, I would think that they could do this fine and if the preconditioner is already on the accelerator, it would be a no-op?

Reply:
The error seems to be that if a preconditioner is already on the accelerator, then calling MoveToAccelerator should be a no-op. A user isnt going to know about the implementation detail that calling it twice causes a memory error. We should be able to do:
 
for(in i = 0; i < 100; i++)
{
    p->MoveToAccelerator();
}
 
and have no errors. Do you agree @ntrost57 ?